### DISCUSSION TITLE
Any plans for C bindings?

### DISCUSSION THREAD

**Discussion by tazz4843** [openai/whisper#125] [Ideas]
I'm very interested in this project. However, my current usage of speech-to-text as a general workload is written in Rust, not Python, so I can't use Whisper without the reasonably large overhead of calling to the CLI every time. It would be great if Whisper were just compiled directly in, without needing to install more stuff on the host. Any plans for C bindings to make this possible?

**Comment by ggerganov** [comment]
This probably is not exactly what you are looking for, but I just released a raw C/C++ implementation of whisper:

https://github.com/ggerganov/whisper.cpp

It's not yet suited for including in other projects but, it might serve as a starting point / reference.

**Comment by FurkanGozukara** [comment]
Perhaps you can call its compiled exe through CMD? I plan to make a video tutorial for this. automating whisper through CMD commands

**Reply by tazz4843** [reply]
I'd prefer to be able to directly use it without having to install it on the host system.
